conu/backend/null/container.py

```diff
@@ -0,0 +1,205 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+Implementation of a nspawn container
+"""
+
+import logging
+import subprocess
+import xmltodict
+import time
+
+from conu.apidefs.container import Container
+from conu.utils import run_cmd
+
+
+logger = logging.getLogger(__name__)
+
+
+class NullContainer(Container):
+
+    def __init__(self, image, container_id, name=None,
+                 popen_instance=None, start_process=None, start_action=None):
+        """
+        Utilities usable for containers
+
+        :param image: image to use (instance of Image class)
+        :param container_id: id of running container (created by Image.run method)
+        :param name: optional name of container
+        :param popen_instance: not used anyhow now
+        :param start_process: subporocess instance with start process
+        :param start_action: set with 4 parameters for starting container
+        """
+        super(NullContainer, self).__init__(image, container_id, name)
+        if not name:
+            self.name = self._id
+        self.popen_instance = popen_instance
+        self.start_process = start_process
+        self.start_action = start_action
+
+    def __repr__(self):
+        # TODO: very similar to Docker method, move to API, this is the proper
+        # way
+        return "%s(image=%s, id=%s)" % (
+            self.__class__, self.image, self.get_id())
+
+    def __str__(self):
+        # TODO: move to API
+        return self.get_id()
+
+    def start(self):
+        self.start_process = NullContainer.internal_run_container(name=self.name, callback_method=self.start_action)
+        return self.start_process
+
+    def get_id(self):
+        """
+        get identifier of container
+        :return: str
+        """
+        return self._id
+
+    def inspect(self, refresh=True):
+        """
+        return cached metadata by default (a convenience method)
+
+        :param refresh: bool, returns up to date metadata if set to True
+        :return: dict
+        """
+        # TODO: move to API defs
+        return self.get_metadata(refresh=refresh)
+
+    def get_metadata(self, refresh=True):
+        """
+        return cached metadata by default
+
+        :param refresh: bool, returns up to date metadata if set to True
+        :return: dict
+        """
+        if refresh or not self._metadata:
+            output = run_cmd(["gdbus", "introspect",
```

Comment:
huh, what?

Reply:
systemd magic :-) unable to use ``systemd status`` for machine parsable output, but able to use ``DBUS`` directly to get all messages and format them as XML and then provide it to user :-)